client/cli/src/config.rs

```diff
@@ -428,7 +433,7 @@ pub trait CliConfiguration: Sized {
 			database: self.database_config(&config_dir, database_cache_size, database)?,
 			state_cache_size: self.state_cache_size()?,
 			state_cache_child_ratio: self.state_cache_child_ratio()?,
-			pruning: self.pruning(is_dev, &role)?,
+			pruning: self.pruning(unsafe_pruning || is_dev, &role)?,
```

Comment:
This should be only `unsafe_pruning`. I have mistake unsafe_pruning and is_dev. In the original code it was not doing this `|| is_dev` at all as you can see:

https://github.com/paritytech/substrate/blob/ddb309ae7c70e5e51a60879af18819cf28be4a32/client/cli/src/lib.rs#L793-L813

Reply:
Create a pr and fix it.

Reply:
https://github.com/paritytech/substrate/pull/5810